File: solution/1002. Find Common Characters/1002.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<int> f(26, INT_MAX);
        for (string& w : words) {
            vector<int> f2(26);
            for (char& c : w)
                f2[c - 'a']++;
            for (int i = 0; i < 26; ++i) {
                f[i] = min(f[i], f2[i]);
            }
        }
        vector<string> res;
        for (int i = 0; i < 26; ++i) {
            while (f[i]--)
                res.push_back(string(1, i + 'a'));
        }
        return res;
    }
};

File: solution/1002. Find Common Characters/1002.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        res = Counter(words[0])
        for w in words:
            res &= Counter(w) # Counter 上的 & 操作會取兩者中的最小值
        return list(res.elements()) # Counter 的 elements() 會幫你按照數目逐個取出來 {a: 3} -> ['a', 'a', 'a']
    
